### ISSUE TITLE
Build Error on Solaris 11

### ISSUE THREAD

**Issue by nxdefiant** [objectcomputing/OpenDDS#293]
Building OpenDDS 3.8 (or git HEAD) fails on Solaris 11.3 directly after invoking gmake. Error below.

Configure options:
`./configure --java=/usr/jdk/instances/jdk1.8.0/`

Error:
```
user@solaris:/opt/OpenDDS-3.8$ gmake
gmake[1]: Entering directory `/opt/OpenDDS-3.8/ACE_wrappers/ace'

GNUmakefile: /opt/OpenDDS-3.8/ACE_wrappers/ace/GNUmakefile.ACE MAKEFLAGS=w

CC -mt -g -library=Cstd  -DACE_HAS_KSTAT -DACE_HAS_CUSTOM_EXPORT_MACROS=0 -D_POSIX_PTHREAD_SEMANTICS   -I/opt/OpenDDS-3.8/ACE_wrappers -DACE_HAS_SCTP -DACE_HAS_LKSCTP -D__ACE_INLINE__ -I.. -DACE_BUILD_DLL  -c -KPIC -o .shobj/Local_Name_Space.o Local_Name_Space.cpp
"/opt/OpenDDS-3.8/ACE_wrappers/ace/config.h", line 1: Error: Could not open include file "ace/config-solaris.h".
"/opt/OpenDDS-3.8/ACE_wrappers/ace/Assert.h", line 23: Error: ACE_Proper_Export_Flag is not defined.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/Assert.h", line 23: Error: A declaration does not specify a tag or an identifier.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/Assert.h", line 23: Error: Use ";" to terminate declarations.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 48: Error: Expected an expression.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 55: Error: Multiple declaration for ACE_Errno_Guard.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 55: Warning: Implicit int is not supported in C++.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 58: Error: The destructor name must match the class name.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 58: Error: Illegal number of arguments for ~file level().
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 66: Error: operator=(int) must be a member function.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 69: Error: Illegal number of arguments for operator==(int).
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 72: Error: Illegal number of arguments for operator!=(int).
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 74: Error: A declaration was expected instead of "private".
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 76: Error: Type name expected instead of "ACE_Errno_Guard".
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 76: Error: Multiple declaration for ACE_Errno_Guard.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 76: Warning: Implicit int is not supported in C++.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 77: Error: Type name expected instead of "ACE_Errno_Guard".
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 77: Error: Type name expected instead of "ACE_Errno_Guard".
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 77: Error: operator=(?&) must be a member function.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.h", line 83: Error: A declaration was expected instead of "}".
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.inl", line 8: Error: ACE_Errno_Guard is not defined.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.inl", line 8: Error: Multiple declaration for ACE_Errno_Guard.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.inl", line 8: Warning: Implicit int is not supported in C++.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.inl", line 10: Error: Only a constructor can have constructor initializers.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.inl", line 19: Warning: "ACE_Errno_Guard(int&, int)" is expected to return a value.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.inl", line 22: Error: ACE_Errno_Guard is not defined.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.inl", line 22: Error: Multiple declaration for ACE_Errno_Guard.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.inl", line 22: Warning: Implicit int is not supported in C++.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.inl", line 23: Error: Only a constructor can have constructor initializers.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.inl", line 29: Warning: "ACE_Errno_Guard(int&)" is expected to return a value.
"/opt/OpenDDS-3.8/ACE_wrappers/ace/OS_Errno.inl", line 32: Error: ACE_Errno_Guard is not defined.
Compilation aborted, too many Error messages.
gmake[1]: *** [.shobj/Local_Name_Space.o] Fehler 2
gmake[1]: Leaving directory `/opt/OpenDDS-3.8/ACE_wrappers/ace'
gmake: *** [ACE] Fehler 2
```

**Comment by mitza-oci** [comment]
Supported platforms (http://opendds.org/documents/building.html) do not currently include Solaris 11.  We don't have this in-house to test with.  So you'll probably need some manual configuration to get it working.  Your config.h should have #include "ace/config-sunos5.11.h" in it.

**Comment by nxdefiant** [comment]
Next Error:

```
gmake[1]: Entering directory `/opt/OpenDDS-3.8/ACE_wrappers/ace'

GNUmakefile: /opt/OpenDDS-3.8/ACE_wrappers/ace/GNUmakefile.ACE MAKEFLAGS=w

CC -mt -g -library=Cstd  -DACE_HAS_KSTAT -DACE_HAS_CUSTOM_EXPORT_MACROS=0 -D_POSIX_PTHREAD_SEMANTICS   -I/opt/OpenDDS-3.8/ACE_wrappers -DACE_HAS_SCTP -DACE_HAS_LKSCTP -D__ACE_INLINE__ -I.. -DACE_BUILD_DLL  -c -KPIC -o .shobj/Local_Name_Space.o Local_Name_Space.cpp
"/usr/include/netinet/sctp.h", line 138: Error: An array cannot have zero size unless you use the option -features=zla.
"/usr/include/netinet/sctp.h", line 581: Error: An array cannot have zero size unless you use the option -features=zla.
```

Strange since netinet/sctp.h is their own Makefile with their own compiler. I'm going to try the zla-suggestion in ACE_wrappers/include/makeinclude/platform_sunos5_sunc++.GNU

**Comment by mitza-oci** [comment]
That seems like a reasonable next step.  However I suggest you work on the ACE port in its own repository https://github.com/DOCGroup/ACE_TAO and open a pull request.  I can help you get it merged (and merge the result into TAO 2.2a).

**Comment by nxdefiant** [comment]
ok, lets see if I can fix this. For anyone interested: Fork is https://github.com/nxdefiant/ACE_TAO
